### PR TITLE
Linting fixes

### LINKED ISSUE #553: Fix React linting warnings.
See https://github.com/nearform/optic-expo/pull/552/files

### PR DESCRIPTION
**Pull request by aledgjones**
closes #553

Added `react/no-unstable-nested-components` exception for allowAsProps as the current usage seems valid based on the api provided by `react-native-paper`.

### CODE REVIEW

**Comment by simoneb** [comment]
what's the status of this PR?

File: src/context/PendingNotificationsContext.tsx [hunk 1]
```diff
@@ -48,8 +48,7 @@ export const PendingNotificationsProvider = ({ children }) => {
       const notificationId = notification.request.content.data.uniqueId
       const notAddedYet =
         pendingNotifications.findIndex(
-          notification =>
-            notification.request.content.data.uniqueId === notificationId
+          item => item.request.content.data.uniqueId === notificationId
```
Comment (simoneb) [inline src/context/PendingNotificationsContext.tsx:51]: why this rename?

File: src/context/SecretsContext.tsx [hunk 2]
```diff
@@ -48,10 +48,10 @@ export const SecretsProvider: React.FC<SecretsProviderProps> = ({
     let mounted = true
     async function initialize() {
       if (user) {
-        const secrets = await secretsManager.getAllByUser(user.uid)
+        const items = await secretsManager.getAllByUser(user.uid)
```
Comment (simoneb) [inline src/context/SecretsContext.tsx:51]: why this rename?

File: src/screens/TokenScreen.tsx [hunk 3]
```diff
@@ -176,19 +175,29 @@ export const TokenScreen = ({ route, navigation }: Props) => {
   // Keep the description for the token up to date
   useEffect(() => {
     const updateDescription = async () => {
-      const tokens = secret.tokens ? [...secret.tokens] : []
-      const existingItemIndex = tokens.findIndex(item => item.token === token)
-      if (existingItemIndex === -1) {
+      const existingTokens = secret.tokens ? [...secret.tokens] : []
```
Comment (simoneb) [inline src/screens/TokenScreen.tsx:178]: I am not convinced that the new implementation is better than the original. In any case, let's please keep this PR to what the original intent is without introducing additional unrelated changes

**Review by simoneb — CHANGES_REQUESTED** [review]

File: src/context/PendingNotificationsContext.tsx (hunk 1 above)
Comment (aledgjones) [reply src/context/PendingNotificationsContext.tsx:51]: Notification is already defined. I'm happy to update to a better name -- I just couldn't think of one!

Comment (simoneb) [reply src/context/PendingNotificationsContext.tsx:51]: generally speaking shadowing variables is not a linting error, is it in this case? if so, fine with this change

Comment (aledgjones) [reply src/context/PendingNotificationsContext.tsx:51]: Yes, it looks like these are showing as linting warnings in the link attached to the issue.

Comment (simoneb) [reply src/context/PendingNotificationsContext.tsx:51]: okay fair enough

File: src/screens/TokenScreen.tsx (hunk 3 above)
Comment (aledgjones) [reply src/screens/TokenScreen.tsx:178]: Understood, I'll revert these changes.

**Comment by simoneb** [comment]
@aledgjones please take out of draft when ready

**Pull request merged**